lean4-infoview/src/infoview/info.tsx

```diff
@@ -375,26 +376,37 @@ function InfoAux(props: InfoProps) {
             setRpcSess(rpcSess0);
             setStatus('ready');
         } catch (ex: any) {
+            let errorString : string;
             if (isRpcError(ex) && ex.code === RpcErrorCode.ContentModified) {
                 // Document has been changed since we made the request, try again
                 void triggerUpdate();
                 return;
             }
-            let errorString : string;
-            if (typeof ex === 'string') {
-                errorString = ex
-            } else if (isRpcError(ex)) {
-                errorString = mapRpcError(ex).message
-            } else if (ex instanceof Error) {
-                errorString = ex.toString()
-            } else if (ex === undefined || JSON.stringify(ex) === '{}')  {
-                // we need to check if this value is empty or not, because maybe we are assigning
-                // a message error with an empty error
-                setError(undefined);
-                return;
-            } else {
-                // unrecognised error
-                errorString = `Unrecognised error: ${JSON.stringify(ex)}`
+            if (isRpcError(ex) && ex.code === RpcErrorCode.InvalidParams) {
+                // Check if the position is valid, if yes just update error message
+                if(pos.line !== undefined || null && pos.line !== undefined || null)
+                {
+                    errorString = 'Invalid Lean code';
+                } else {
+                    errorString = mapRpcError(ex).message
+                }
```

Comment:
I think this could instead be moved to mapRpcError in Util.ts, then this file would not need to change.

Reply:
Ok, good point!